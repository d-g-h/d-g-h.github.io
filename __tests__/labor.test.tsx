/**
 * @jest-environment jsdom
 */
import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import Labor from "@/components/labor";

const KEY = "0";
const COMPANY = "Company";
const TITLE = "Job Title";
const LOCATION = "Location";
const START = "2020-02";
const END = "2024-02";

describe("Labor", () => {
  it("renders Labor", () => {
    render(
      <Labor
        key={KEY}
        title={TITLE}
        location={LOCATION}
        company={COMPANY}
        start={START}
        end={END}
      >
        <></>
      </Labor>,
    );

    const jobTitle = screen.getByText("Job Title");
    const location = screen.getByText("Location");

    expect(jobTitle).toBeInTheDocument();
    expect(location).toBeInTheDocument();
  });
  it("renders Labor without Company", () => {
    render(
      <Labor
        key={KEY}
        title={TITLE}
        location={LOCATION}
        start={START}
        end={END}
      >
        <></>
      </Labor>,
    );

    const jobTitle = screen.getByText("Job Title");
    const company = screen.queryByText("Company");

    expect(jobTitle).toBeInTheDocument();
    expect(company).toBeNull();
  });
  it("renders html correctly", () => {
    const { container } = render(
      <Labor
        key={KEY}
        title={TITLE}
        location={LOCATION}
        company={COMPANY}
        start={START}
        end={END}
      >
        <></>
      </Labor>,
    );
    expect(container).toMatchSnapshot();
  });
});
