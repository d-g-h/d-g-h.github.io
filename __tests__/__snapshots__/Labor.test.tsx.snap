// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Labor renders html correctly 1`] = `
<div>
  <li
    class="labor"
  >
    <div
      class="title"
    >
      Job Title
    </div>
    <div
      class="highlight"
    >
      Company
    </div>
    <div
      class="location"
    >
      Location
    </div>
    <div
      class="time"
    >
      <time
        datetime="2020-02"
      >
        February 2020
      </time>
      <span>
         - 
      </span>
      <time
        datetime="2024-02"
      >
        February 2024
      </time>
    </div>
  </li>
</div>
`;
